pkg/runtimes/docker/container.go

```diff
@@ -176,3 +176,62 @@ func getNodeContainer(ctx context.Context, node *k3d.Node) (*types.Container, er
 	return &containers[0], nil
 
 }
+
+// executes an arbitrary command in a container while returning its exit code.
+// useful to check something in docker env
+func executeCheckInContainer(ctx context.Context, image string, cmd []string) (int64, error) {
+	docker, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		log.Errorln("Failed to create docker client")
+		return -1, err
+	}
+	defer docker.Close()
+
+	if err = pullImage(ctx, docker, image); err != nil {
+		return -1, err
+	}
+
+	resp, err := docker.ContainerCreate(ctx, &container.Config{
+		Image:      image,
+		Cmd:        cmd,
+		Tty:        false,
+		Entrypoint: []string{},
+	}, nil, nil, nil, "")
+	if err != nil {
+		log.Errorf("Failed to create container from image %s with cmd %s", image, cmd)
+		return -1, err
+	}
+
+	if err = startContainer(ctx, resp.ID); err != nil {
+		return -1, err
+	}
+
+	exitCode := -1
+	statusCh, errCh := docker.ContainerWait(ctx, resp.ID, container.WaitConditionNotRunning)
+	select {
+	case err := <-errCh:
+		if err != nil {
+			log.Errorf("Error while waiting for container %s to exit", resp.ID)
+			return -1, err
+		}
+	case status := <-statusCh:
+		exitCode = int(status.StatusCode)
+	}
+
+	if err = removeContainer(ctx, resp.ID); err != nil {
+		return -1, err
+	}
+
+	return int64(exitCode), nil
+}
+
+// CheckIfDirectoryExists checks for the existence of a given path inside the docker environment
+func CheckIfDirectoryExists(ctx context.Context, image string, dir string) (bool, error) {
```

Comment:
I might be missing something here, but this only validates, if that path exists inside the chosen container's/image's filesystem, doesn't it? What is this required for here?

Reply:
yes, it checks for the existence of a path in a given container. In this implementation it is required to check if EDAC folder in particular exists inside the container as this makes it trivial to be portable across platforms (linux, mac, windows).
The whole function to check is abstracted away and kept here as it is general and can be used for other stuff but can also be moved from this module if you see a better place for it.

Reply:
I was just wondering, why it's testing with a new container (given an image name as input) instead of exec'ing inside the running container :thinking:

Reply:
to clarify: it is because we need to check if this folder exists before running the node, as we need to know whether edac folder exists or not - if its not there, docker will throw an error if we try mounting